### PR TITLE
[Python] Separate out notebooks for tensorflow with tfx and built-in model handler

### PR DESCRIPTION
**Pull request by riteshghorse**
Different notebooks to demonstrate `tfx-bsl`'s model handler and `TFModelHandlerNumpy`. Majority of the content is same as earlier notebook.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by riteshghorse** [comment]
R: @rezarokni @rszper 
CC: @damccorm

File: examples/notebooks/beam-ml/run_inference_tensorflow.ipynb [hunk 1]
```diff
@@ -67,21 +67,18 @@
...
+        "Beam has built in support for 2 TensorFlow Model Handlers: [TFModelHandlerNumpy](https://github.com/apache/beam/blob/ca0787642a6b3804a742326147281c99ae8d08d2/sdks/python/apache_beam/ml/inference/tensorflow_inference.py#L91) and [TFModelHandlerTensor](https://github.com/apache/beam/blob/ca0787642a6b3804a742326147281c99ae8d08d2/sdks/python/apache_beam/ml/inference/tensorflow_inference.py#L184).\n",
+        "TFModelHandlerNumpy can be used to run inference on models expecting a `numpy` array as an input while TFModelHandlerTensor can be used to run inference on models expecting a `tf.Tensor` as an input.\n",
         "\n",
-        "Beam's Runinference transform also accepts a ModelHandler generated from [`tfx-bsl`](https://github.com/tensorflow/tfx-bsl) using `CreateModelHandler`.\n",
+        "If your model needs input of type `tf.Example` then please look at this [notebook](https://colab.research.google.com/github/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb).\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow.ipynb:73]:
```suggestion
        "If your model needs input of type `tf.Example` see the [Apache Beam RunInference with `tfx-bsl` notebook](https://colab.research.google.com/github/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb).\n",
```

File: examples/notebooks/beam-ml/run_inference_tensorflow.ipynb [hunk 2]
```diff
@@ -90,19 +87,7 @@
...
-      ],
-      "metadata": {
-        "id": "HrCtxslBGK8A"
-      }
-    },
-    {
-      "cell_type": "markdown",
-      "source": [
-        "To use RunInference with built-in Tensorflow Model Handler, install Apache Beam version 2.46 or later. Creation of a ModelHandler is supported in `tfx-bsl` versions 1.10 and later."
+        "To use RunInference with built-in Tensorflow Model Handler, install Apache Beam version 2.46 or later."
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow.ipynb:90]:
```suggestion
        "To use RunInference with built-in Tensorflow model handler, install Apache Beam version 2.46 or later."
```

File: examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+        "  </td>\n",
+        "</table>\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "This notebook demonstrates the use of the [RunInference](https://beam.apache.org/releases/pydoc/current/apache_beam.ml.inference.base.html#apache_beam.ml.inference.base.RunInference) transform for [TensorFlow](https://www.tensorflow.org/) using [tfx-bsl](https://github.com/tensorflow/tfx-bsl).\n",
+        "\n",
+        "This approach should only be used when the trained model requires a `tf.Example` input. If you have `numpy` or `tf.Tensor` inputs, please look at this [notebook](https://colab.research.google.com/github/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_tensorflow.ipynb) instead which demonstrates the use built-in model handlers for TensorFlow in Beam SDK starting 2.46.0.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb:70]:
```suggestion
        "Use this approach when the trained model requires a `tf.Example` input. If you have `numpy` or `tf.Tensor` inputs, see the [Apache Beam RunInference with TensorFlow notebook](https://colab.research.google.com/github/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_tensorflow.ipynb) instead which demonstrates the use built-in model handlers for TensorFlow in Beam SDK starting 2.46.0.\n",
```

File: examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "A8xNRyZMW1yK"
+      },
+      "source": [
+        "# Apache Beam RunInference with TensorFlow\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb:53]: The notebooks shouldn't have the same title. Maybe change this one to:

Apache Beam RunInference with tfx-bsl

File: examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "This notebook demonstrates the use of the [RunInference](https://beam.apache.org/releases/pydoc/current/apache_beam.ml.inference.base.html#apache_beam.ml.inference.base.RunInference) transform for [TensorFlow](https://www.tensorflow.org/) using [tfx-bsl](https://github.com/tensorflow/tfx-bsl).\n",
+        "\n",
+        "This approach should only be used when the trained model requires a `tf.Example` input. If you have `numpy` or `tf.Tensor` inputs, please look at this [notebook](https://colab.research.google.com/github/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_tensorflow.ipynb) instead which demonstrates the use built-in model handlers for TensorFlow in Beam SDK starting 2.46.0.\n",
+        "\n",
+        "Beam's Runinference transform accepts a ModelHandler generated from [`tfx-bsl`](https://github.com/tensorflow/tfx-bsl) using `CreateModelHandler`.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb:72]:
```suggestion
        "The Apache Beam RunInference transform accepts a model handler generated from [`tfx-bsl`](https://github.com/tensorflow/tfx-bsl) by using `CreateModelHandler`.\n",
```

File: examples/notebooks/beam-ml/run_inference_tensorflow.ipynb [hunk 6]
```diff
@@ -67,21 +67,18 @@
...
+        "TFModelHandlerNumpy can be used to run inference on models expecting a `numpy` array as an input while TFModelHandlerTensor can be used to run inference on models expecting a `tf.Tensor` as an input.\n",
         "\n",
-        "Beam's Runinference transform also accepts a ModelHandler generated from [`tfx-bsl`](https://github.com/tensorflow/tfx-bsl) using `CreateModelHandler`.\n",
+        "If your model needs input of type `tf.Example` then please look at this [notebook](https://colab.research.google.com/github/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb).\n",
         "\n",
         "The Apache Beam RunInference transform is used to make predictions for\n",
         "a variety of machine learning models. For more information about the RunInference API, see [Machine Learning](https://beam.apache.org/documentation/sdks/python-machine-learning) in the Apache Beam documentation.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow.ipynb:76]:
```suggestion
        "a variety of machine learning models. For more information about the RunInference API, see [Get started with AI/ML pipelines](https://beam.apache.org/documentation/ml/overview/) in the Apache Beam documentation.\n",
```

We should update this link.

File: examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb [hunk 7]
```diff
@@ -0,0 +1,647 @@
...
+        "\n",
+        "The Apache Beam RunInference transform is used to make predictions for\n",
+        "a variety of machine learning models. For more information about the RunInference API, see [Machine Learning](https://beam.apache.org/documentation/sdks/python-machine-learning) in the Apache Beam documentation.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb:75]:
```suggestion
        "a variety of machine learning models. For more information about the RunInference API, see [Get started with AI/ML pipelines](https://beam.apache.org/documentation/ml/overview/) in the Apache Beam documentation.\n",
```

File: examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb [hunk 8]
```diff
@@ -0,0 +1,647 @@
...
+        "\n",
+        "This notebook demonstrates the following steps:\n",
+        "- Import [`tfx-bsl`](https://github.com/tensorflow/tfx-bsl).\n",
+        "- Build a simple TensorFlow model.\n",
+        "- Set up example data\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb:80]:
```suggestion
        "- Set up example data.\n",
```

File: examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb [hunk 9]
```diff
@@ -0,0 +1,647 @@
...
+        "- Run those examples with `tfx-bsl` model handler and get a prediction inside an Apache Beam pipeline."
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb:81]:
```suggestion
        "- Run those examples with the `tfx-bsl` model handler and get a prediction inside an Apache Beam pipeline."
```

File: examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb [hunk 10]
```diff
@@ -0,0 +1,647 @@
...
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## RunInference with Tensorflow using tfx-bsl\n",
+        "In versions 1.10.0 and later of `tfx-bsl`, you can\n",
+        "create a TensorFlow `ModelHandler` for use with Apache Beam. For more information about the RunInference API, see [Machine Learning](https://beam.apache.org/documentation/sdks/python-machine-learning) in the Apache Beam documentation.\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb:307]:
```suggestion
        "create a TensorFlow `ModelHandler` for use with Apache Beam. For more information about the RunInference API, see [Get started with AI/ML pipelines](https://beam.apache.org/documentation/ml/overview/) in the Apache Beam documentation.\n",
```

File: examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb [hunk 11]
```diff
@@ -0,0 +1,647 @@
...
+            "example is 60.00 prediction is 300.87\n",
+            "example is 90.00 prediction is 448.26\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## KeyedModelHandler with TensorFlow using tfx-bsl\n",
```
Comment (rszper) [inline examples/notebooks/beam-ml/run_inference_tensorflow_with_tfx.ipynb:568]:
```suggestion
        "## KeyedModelHandler with TensorFlow using `tfx-bsl`\n",
```

**Comment by riteshghorse** [comment]
Thanks for the review! Addressed the comments.

**Comment by riteshghorse** [comment]
Run RAT PreCommit

**Comment by riteshghorse** [comment]
Run RAT PreCommit

**Comment by rszper** [comment]
LGTM

**Comment by damccorm** [comment]
@riteshghorse FYI this may end up getting reverted because of the current code freeze - https://lists.apache.org/thread/rpmq9r9tbgrry8yco70mzyt4psyk5m20 - if that happens we can just make the same change again (I'll be happy to approve)

**Comment by riteshghorse** [comment]
Oops sorry! I missed that email, revert PR at https://github.com/apache/beam/pull/26143

**Pull request merged**